tests/unit_tests/webapp/utils/test_callbacks.py

```diff
@@ -0,0 +1,528 @@
+        self.df = df
+
+        dataframe_x = df[['column1','column3']].copy()
+        y_target = pd.DataFrame(data=np.array([0, 0, 0, 1, 1]), columns=['pred'])
+        model = DecisionTreeClassifier().fit(dataframe_x, y_target)
+        features_dict = {'column3': 'Useless col'}
+        additional_data = df[['column2','column4','column5']].copy()
+        additional_features_dict = {'column2': 'Additional col'}
+        self.xpl = SmartExplainer(model=model, features_dict=features_dict)
+        self.xpl.compile(
+            x=dataframe_x, 
+            y_pred=y_target,
+            y_target=y_target,
+            additional_data=additional_data, 
+            additional_features_dict=additional_features_dict
+        )
+        self.smart_app = SmartApp(self.xpl)
+
+        self.click_data = {
+            'points': [
+                {
+                    'curveNumber': 0, 
+                    'pointNumber': 3, 
+                    'pointIndex': 3, 
+                    'x': 0.4649, 
+                    'y': 'Sex', 
+                    'label': 'Sex', 
+                    'value': 0.4649, 
+                    'customdata': 'Sex', 
+                    'marker.color': 'rgba(244, 192, 0, 1.0)', 
+                    'bbox': {'x0': 717.3, 'x1': 717.3, 'y0': 82.97, 'y1': 130.78}
+                }
+            ]
+        }
+        self.special_cols = ['_index_', '_predict_', '_target_']
+
+        super(TestCallbacks, self).__init__(*args, **kwargs)
+
+    def test_default_init_data(self):
+        expected_result = pd.DataFrame(
+            {
+                '_index_': [0, 1, 2, 3, 4],
+                '_predict_': [0, 0, 0, 1, 1],
+                '_target_': [0, 0, 0, 1, 1],
+                'column1': [1, 2, 3, 4, 5],
+                'column3': [1.1, 3.3, 2.2, 4.4, 5.5],
+                '_column2': ['a', 'b', 'c', 'd', 'e'],
+                '_column4': [True, False, True, False, False],
+                '_column5': pd.date_range('2023-01-01', periods=5),
+            },
+        )
+        self.smart_app.init_data()
+        pd.testing.assert_frame_equal(expected_result, self.smart_app.round_dataframe)
+    
+    def test_limited_rows_init_data(self):
+        self.smart_app.init_data(3)
+        assert len(self.smart_app.round_dataframe)==3
+
+    def test_select_data_from_prediction_picking(self):
+        selected_data = {"points": [{"customdata":0}, {"customdata":2}]}
+        expected_result = pd.DataFrame(
+            {
+                'column1': [1, 3],
+                'column2': ['a', 'c'],
+                'column3': [1.1, 2.2],
+                'column4': [True, True],
+                'column5': [pd.Timestamp('2023-01-01'), pd.Timestamp('2023-01-03')],
+            },
+            index=[0, 2]
+        )
+        result = select_data_from_prediction_picking(self.df, selected_data)
+        pd.testing.assert_frame_equal(expected_result, result)
+    
+    def test_select_data_from_filters_string(self):
```

Comment:
The readibility of all these test_select_data_from_filters_*** tests would benefit from splitting the select_data_from_filters and reducing the number of args.